app/controllers/api/buildings_controller.rb

```diff
@@ -0,0 +1,33 @@
+class Api::BuildingsController < ApplicationController
+
+  def index
+    @buildings = Building.all
+    render json: @buildings
+  end
+
+  def create
+    # should always create buildings with a portfolio
+    @portfolio = Portfolio.find(params[:portfolio_id])
+    @building = @portfolio.buildings.create(building_params)
+
+    if @building.save
+      render_json_message(:ok, data: @building, message: 'Successful Creation and Save')
```

Comment:
can mention that it was a building that was successfully created and save?
'Building {id} successfully created and saved'